addons/knowledge/static/src/js/knowledge_controller.js

```diff
@@ -106,6 +107,29 @@ const KnowledgeArticleFormController = FormController.extend({
         await this._rename(id, event.currentTarget.value);
     },
 
+    /**
+     * When the user clicks on the name of the article, checks if the article
+     * name hasn't been set yet. If it hasn't, it will look for a title in the
+     * body of the article and set it as the name of the article.
+     * @param {Event} event
+     */
+    _onSetArticleName: async function (event) {
```

Comment:
Still posting some guidelines even though this will be converted to an onchange if I followed correctly.

Method should be named "_onArticleBreadcrumbClick" or something along those lines.
Currently it's named "_onSetArticleName", which makes you think this is called when the article name is set, which is not the case.